lib/ts/recipeImplementation.ts

```diff
@@ -97,7 +97,24 @@ export default function RecipeImplementation(recipeImplInput: {
         },
         doesSessionExist: async function(_: { userContext: any }): Promise<boolean> {
             logDebugMessage("doesSessionExist: called");
-            return (await getIdRefreshToken(true)).status === "EXISTS";
+
+            const tokenInfo = await FrontToken.getTokenInfo();
+
+            // The above includes getIdRefreshToken(true), which would call refresh if the FE cookies were cleared for some reason
+            if (tokenInfo === undefined) {
+                logDebugMessage("doesSessionExist: access token does not exist locally");
+                return false;
+            }
+
+            if (tokenInfo.ate < Date.now()) {
+                logDebugMessage("doesSessionExist: access token expired. Refreshing session");
+
+                const preRequestIdToken = await getIdRefreshToken(false);
+                const refresh = await onUnauthorisedResponse(preRequestIdToken);
+                return refresh.result === "RETRY";
```

Comment:
See how refreshing is done in getAccessTokenPayloadSecurely in this file. Do it in a similar way, except that instead of `throw new Error("Could not refresh session");` we should return `false`?

Reply:
This is what I did here; I just replaced the `attemptRefreshingSession` since it throws if refreshing fails.